bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java

```diff
@@ -303,22 +369,28 @@ public String getCategoryIdentifier(SlingHttpServletRequest request) {
             return identifier;
         }
 
-        CategoryUrlFormat.Params categoryIdentifiers = newCategoryUrlFormat
-            .parse(request.getRequestPathInfo(), request.getRequestParameterMap());
+        CategoryUrlFormat.Params categoryIdentifiers = getUrlFormatFromContext(request, null,
```

Comment:
For better readability, please get the format first and than call `parse()` on it.